### ISSUE TITLE
Using across in group_by on previously grouped data exhibits strange behavior

### ISSUE THREAD

**Issue by caldwellst** [tidyverse/dplyr#5253]
When using `across()` within `group_by()` on a data frame, the behavior isn't as expected if variables selected through `across()` are existing grouping columns of the data frame. This seems to be because the column selection seems to exclude grouped columns by default. Because of this, `group_by()` using `across()` mimics more the behavior of `mutate_at()` rather than `group_by_at()`.

```r
## works
mtcars %>%
  group_by(across(c(mpg:hp)))
#> # A tibble: 32 x 11
#> # Groups:   mpg, cyl, disp, hp [31]

## works since drat not in across selection
mtcars %>%
  group_by(drat) %>%
  group_by(across(c(mpg:hp)))
#> # A tibble: 32 x 11
#> # Groups:   mpg, cyl, disp, hp [31]

mtcars %>%
  group_by(hp) %>%
  group_by(across(c(mpg:hp)))
#> Note: Using an external vector in selections is ambiguous.
#> i Use `all_of(hp)` instead of `hp` to silence this message.
#> i See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.
#> This message is displayed once per session.
#> Error: Problem with `mutate()` input `..1`.
#> x Can't subset columns that don't exist.
#> x Locations 11, 12, 13, 14, 15, etc. don't exist.
#> i There are only 10 columns.
#> i Input `..1` is `across(c(mpg:hp))`.
#> i The error occured in group 1: hp = 52.
```
When using selection helpers, the grouping will just silently exclude previously grouping variables, unless the selection is empty.
```r
mtcars %>%
  group_by(hp) %>%
  group_by(across(contains("p")))
#> # A tibble: 32 x 11
#> # Groups:   mpg, disp [31]

mtcars %>%
  group_by(across(c(mpg:hp))) %>%
  group_by(across(contains("p")))
#> Error: `vars` must be a character vector.
```

**Comment by hadley** [comment]
I'd say this is by design; `across()` works the same way in all verbs.

**Comment by caldwellst** [comment]
Okay. From my experience, I think could be easy to expect behavior similar to `group_by_at()` and not notice the differneces. Could it be useful to add a bit to the colwise vignette just to reiterate the point on across not selecting grouping variables and implications for `group_by()`? Happy to submit a PR if so.

**Comment by eipi10** [comment]
Regardless of whether this is by design, it might be unexpected for some (many?) users, including me. Consistency is often a virtue, but rigid consistency across *all* contexts sometimes gives results that don't meet preexisting expectations in *some* contexts. It would be helpful to have an informative error message about the behavior of `across` within `group_by`. 

For example, I just got the error message below when trying `group_by` and `across` on a data frame that was already grouped by the first two listed columns:

```
d %>%  group_by(across(c(`I/E`, fiscal.year, matches("acct"))))

Error: Problem with `mutate()` input `..1`.
x Can't subset columns that don't exist.
x Column `I/E` doesn't exist.
ℹ Input `..1` is `across(c(`I/E`, fiscal.year, matches("acct")))`.
ℹ The error occurred in group 1: I/E = "Expense", fiscal.year = "2018-19", acct4 = "500".
Run `rlang::last_error()` to see where the error occurred.
```

I found this completely baffling until I stumbled across this issue page.
